core/src/main/java/org/apache/iceberg/actions/BaseExpireSnapshotsActionResult.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+public class BaseExpireSnapshotsActionResult implements ExpireSnapshots.Result {
```

Comment:
I'm grateful that the  ExpireSnapshotsAction refactoring work has been considered to work for different compute engines, such as  flink.  In this way,  we flink don't have to abstract the common logic of actions between flink and spark and we could just extend those BaseXXAction to implement our flink own actions.  That's really important ! 

About the `ExpireSnapshotsAction`,  I think both flink & spark will have the same `Result`.  So maybe we could just use the `ExpireSnapshotsActionResult` directly ( I don't think there will be a reason that flink or spark will extend this `BaseExpireSnapshotsActionResult`, so maybe we could just remove the `Base` prefix).

Reply:
Here, I followed the pattern we have in many classes in `core` like `BaseFile`, `BaseTable`, `BaseRewriteManifests`. Also, there is a class called `ExpireSnapshotsActionResult` in this package in Spark already. We deprecate it with the introduction of this new one.

I agree query engines will most likely use the same result classes.

Reply:
What about making `ExpireSnapshots.Result` a class instead of an interface? Then we would always have the same default implementation.

Reply:
I am fine converting `Result` into a class but I am not sure we will gain much by doing that. The current class is in the core module so it is accessible to everyone. Making `Result` a class may make inheritance a bit harder. I do like our current pattern of having `BaseXXX` classes in core and interfaces in the API module.